Cython/Utility/ObjectHandling.c

```diff
@@ -1992,6 +1992,26 @@ typedef struct {
 /////////////// UnpackUnboundCMethod ///////////////
 //@requires: PyObjectGetAttrStr
 
+#define __PYX_REINTERPRET_FUNCION(func_pointer, other_pointer) ((func_pointer)(void(*)(void))(other_pointer))
+#define __PYX_REINTERPRET_POINTER(pointer_type, pointer) ((pointer_type)(void *)(pointer))
+#define __PYX_RUNTIME_REINTERPRET(type, var) (*(type *)(&var))
+
+static PyObject *__Pyx_SelflessCall(PyObject *method, PyObject *args, PyObject *kwargs) {
+    PyObject *selfless_args = PyTuple_GetSlice(args, 1, PyTuple_Size(args));
```

Comment:
Seems worth using a vectorcall here, if available, rather than creating a new tuple just for the call. (Not sure if that's an optimisation that we need to do right now.)

Reply:
This is definitely an obscure corner case so I'd be inclined to keep it simple for now. Much as this feels like a place that vectorcall would be good for

Reply:
I added a comment with such a note, I think it's sufficient right now